storage/client/src/test/java/com/azure/storage/file/FileServiceClientAsyncTests.java

```diff
@@ -0,0 +1,473 @@
+// Licensed under the MIT License.
+
+package com.azure.storage.file;
+
+import com.azure.core.http.HttpClient;
+import com.azure.core.http.policy.HttpLogDetailLevel;
+import com.azure.storage.file.models.CorsRule;
+import com.azure.storage.file.models.FileServiceProperties;
+import com.azure.storage.file.models.ListSharesOptions;
+import com.azure.storage.file.models.Metrics;
+import com.azure.storage.file.models.RetentionPolicy;
+import com.azure.storage.file.models.ShareItem;
+import com.azure.storage.file.models.ShareProperties;
+import com.azure.storage.file.models.StorageErrorException;
+import org.junit.Assert;
+import reactor.test.StepVerifier;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.Assert.fail;
+
+public class FileServiceClientAsyncTests extends FileServiceClientTestsBase {
+    private FileServiceAsyncClient serviceClient;
+
+    private String reallyLongString = "thisisareallylongstringthatexceedsthe64characterlimitallowedoncertainproperties";
+
+    @Override
+    public void beforeTest() {
+        shareName = getShareName();
+
+        if (interceptorManager.isPlaybackMode()) {
+            serviceClient = setupClient((connectionString, endpoint) -> FileServiceAsyncClient.builder()
+                .connectionString(connectionString)
+                .endpoint(endpoint)
+                .httpClient(interceptorManager.getPlaybackClient())
+                .httpLogDetailLevel(HttpLogDetailLevel.BODY_AND_HEADERS)
+                .buildAsync());
+        } else {
+            serviceClient = setupClient((connectionString, endpoint) -> FileServiceAsyncClient.builder()
+                .connectionString(connectionString)
+                .endpoint(endpoint)
+                .httpClient(HttpClient.createDefault().wiretap(true))
+                .httpLogDetailLevel(HttpLogDetailLevel.BODY_AND_HEADERS)
+                .addPolicy(interceptorManager.getRecordPolicy())
+                .buildAsync());
+        }
+    }
+
+    @Override
+    public void afterTest() {
+        serviceClient.listShares(new ListSharesOptions().prefix(shareName))
+            .collectList()
+            .block()
+            .forEach(share -> {
+                ShareAsyncClient client = serviceClient.getShareAsyncClient(share.name());
+                try {
+                    client.delete(null).block();
+                } catch (StorageErrorException ex) {
+                    // Share already deleted, that's what we wanted anyways.
+                }
+            });
+    }
+
+    @Override
+    public void getShareDoesNotCreateAShare() {
+        ShareAsyncClient client = serviceClient.getShareAsyncClient(shareName);
+        try {
+            client.getStatistics().block();
+            fail("getShareAsyncClient shouldn't create a share in Azure.");
```

Comment:
Why we fail the tests here? Are you trying to get exception from client.getStatistics, can we catch StorageException directly?

Reply:
I have it failing in case it succeeds somehow, I catch Exception instead of StorageException in case another exception type is thrown incorrectly.

Reply:
The purpose of the tests is to catch specific error, not general case. Based on my understanding, the test is suppose to pass. I don't think you need to fail and catch error here.

Reply:
It is better to assert some successful status.